### DISCUSSION TITLE
snap package

### DISCUSSION THREAD

**Discussion by lemmy04** [XEphem/XEphem#4] [General]
Hi,

I've started working on a snap package... but I've never done that before, so any advice will be very welcome.

cheers
[L]

**Comment by brandon-rhodes** [comment]
Alas, my only experience with packing was contributing a few RPMs to the Red Hat community back in the 1990s. I had almost forgotten about it until last month when I discovered that my web page listing them is still on my site (!):

https://rhodesmill.org/brandon/rpms

But I don't know specifically about snaps. I have done a few recent Dockerfiles — you can see one I’ve just built for XEphem here:

https://github.com/XEphem/XEphem/tree/docker/container

The main advice I know from Docker is to minimize container size by not insisting on carrying a full operating system. Some folks are tempted to build atop large distributions like Ubuntu, but there are both much smaller distributions that specialize in small size for containers, and also the possibility of making a container of just the binary and libraries you need, leaving out things like a shell; I remember there are language-specific docker images, like Python images that only have Python but not much else. It can make an image much simpler to build and distribute.

Anyway, thanks for your interest, let us know if you get something built!

**Reply by obspsr** [reply]
Hi Brandon,
I use rpm packages for 20 years and for 5-6 years some debian apt packages with great pleasure ...
But starting a brand new powerful laptop, I burned my fingers with emacs and netbeans "snaps" installation
... no access to netbeans/etc config files in root !! Damned ! (I had not RTFM ...)
So  (just my advice) I agree with you for using mainly "git - cmake - and .tgz"  classical tools...

Maybe out of place, but I want to forward to you that I am on the way to obtain the moons polynome from IMCCE
Mr. J.H. Arlot as I told else where to Elwood and Mathias.
Best regards, Serge.

**Reply by brandon-rhodes** [reply]
> I am on the way to obtain the moons polynome from IMCCE

By reading from a `.bsp` file, or through another mechanism?

**Reply by obspsr** [reply]
I asked him to put the same format here, he did that before .... here :
https://ftp.imcce.fr/pub/misc/satxyz/
Good night ...

**Reply by brandon-rhodes** [reply]
Thanks for the links! Can those files be dropped in as simple replacements for the files in the auxil directory, or do they require source code to be adjusted because, for example, they covered a different range of dates than the code expects?

**Reply by obspsr** [reply]
As soon as the files will appear with extension .2030 ... we will just have to drop them in auxil directory.
Then, in libastro :   # grep -n **use_bdl**    to show files using these data and finish updating the date range and file name to match.
Calls are in libastro/jupmoon.c, marsmoon.c, satmoon.c and umoon.c.
Best regards,
Serge
